### ISSUE TITLE
TokenRefresher error handling

### ISSUE THREAD

**Issue by phillebaba** [Azure/azure-storage-blob-go#303]
### Which version of the SDK was used?
v0.14.0

### Which platform are you using? (ex: Windows, Linux, Debian)
Linux

### What problem was encountered?
Currently the `TokenRefresher`  function passed when creating a NewToken does not have a clear method of dealing with errors when a token refresh fails.
https://github.com/Azure/azure-storage-blob-go/blob/da3335192ee9609890f4172c43d0750c65941b6f/azblob/zc_credential_token.go#L34

Some documentation recommends returning a 0 to stop the token refresh all together, but this is not really an optimal solution. I have seen scenarios where Azure API rate limits have been reached where it would be better to just retry later in an attempt to wait it out. Either way I feel like there should be a better solution to the token refresh logic. 

The point of creating this issue is to start a discussion about a better solution, or understand why the way it is done today is the most optimal solution.

### How can we reproduce the problem in the simplest way?
N/A

### Have you found a mitigation/solution?
N/A

**Comment by MaurGi** [comment]
I agree with this - returning 0 is not a good option, eventually the token will expire and no one will evel refresh it again - so retrying is a good idea - but it should be retrying only for a limited time, then panic/bail because it cannot be refreshed
